user/api.go

```diff
@@ -714,53 +651,78 @@ func (a *Api) sendError(res http.ResponseWriter, statusCode int, reason string,
 	sendModelAsResWithStatus(res, status.NewStatus(statusCode, reason), statusCode)
 }
 
-func (a *Api) authenticateSessionToken(sessionToken string) (*TokenData, error) {
-	if sessionToken == "" {
+func (a *Api) authenticateToken(r *http.Request) (*TokenData, error) {
+	if sessionToken := r.Header.Get(TP_SESSION_TOKEN); sessionToken != "" {
+		return a.authenticateSessionToken(sessionToken)
+	}
+	return a.accessTokenChecker.Check(r)
```

Comment:
Do we want to get priority to the Tidepool Session Token (which effectively lasts forever) or the Auth0 token. I would think we'd want to move to the Auth0 token sooner rather than later so would these checks be reversed?